databricks/sdk/runtime/__init__.py

```diff
@@ -86,23 +86,87 @@ def inner() -> Dict[str, str]:
         _globals[var] = userNamespaceGlobals[var]
     is_local_implementation = False
 except ImportError:
-    from typing import cast
-
     # OSS implementation
     is_local_implementation = True
 
+    try:
+        # We expect this to fail and only do this for providing types
+        from pyspark.sql.context import SQLContext
+        sqlContext: SQLContext = None # type: ignore
+        sql = sqlContext.sql
+        table = sqlContext.table
+    except Exception:
+        pass
+
+    # The next few try-except blocks are for initialising globals in a best effort
+    # mannaer. We separate them to try to get as many of them working as possible
+    try:
+        from pyspark.sql.functions import udf  # type: ignore
+    except ImportError as e:
+        logging.debug(f"Failed to initialise udf global: {e}")
+
+    try:
+        from databricks.connect import DatabricksSession  # type: ignore
+        spark = DatabricksSession.builder.getOrCreate()
+        sc = spark.sparkContext
+    except Exception as e:
+        # We are ignoring all failures here because user might want to initialize
+        # spark session themselves and we don't want to interfere with that
+        logging.debug(f"Failed to initialize spark session: {e}")
```

Comment:
in these debug messages, let's be explicit about which globals were not defined, but also indicate that it wasn't a problem.
```suggestion
        logging.debug(f"Failed to initialize globals 'spark' and 'sc', continuing. Cause: {e}")
```